### PR TITLE
Add documentation file structure and CI

### PR DESCRIPTION
**Pull request by joaomcteixeira**
You are about to submit a new Pull Request. Before continuing, make sure you read the [contributing guidelines](CONTRIBUTING.md) and you comply with the following criteria:

- [x] Your PR is about writing documentation for already existing code :fire:
- [ ] Your PR is about writing tests for already existing code :godmode:

---

Add infrastructure to build interactive documentation webpages. The documentation is a mixture of stand-alone documents written in Markdown (in `docs/` folder) and code documentation written directly in the Python files (in [reStructuredText](https://en.wikipedia.org/wiki/ReStructuredText), and [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html)). With [Sphinx](https://www.sphinx-doc.org/en/master/), everything becomes integrated into a single webpage. We can later host the resulting pages in [ReadTheDocs](https://readthedocs.org/).  **Note:** even if at later stages of the project we decide to move parts of the documentation (for example tutorials) to the group's main website, the current structure allows us to build and organize the docs properly.

To build the documentation HTML pages **locally**:

```
tox -e docs
```

With your browser, open the file: `dists/docs/index.html`. Navigate around and enjoy/comment/contribute.

Once accepted, this building process will become part of the CI pipelines.

## Contributing to docs

For example, to write about the `rigidbody` module, edit the main docstrings in `haddock.modules.sampling.rigidbody.__init__.py`.

### CODE REVIEW

**Review by douweschulte — APPROVED** [review]
Nice addition! Especially running it as one of the commit checks (and failing if some docstrings are erroneous) is really nice.

File: CITING.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Citing
+
+Cite HADDOCK3 with the following:
+
+HADDOCK3, Bonvin's Lab, https://github.com/haddocking/haddock3, 2022.
```
Comment (amjjbonvin) [inline CITING.md:5]: I guess we should give at least the original HADDOCK publication for the time being. I.e the JACS 2003 one, next to the above line.

Comment (amjjbonvin) [reply CITING.md:5]: And may-be also add `Developed with financial support from the Horizon 2020 BioExcel CoE project [823830](https://cordis.europa.eu/project/id/823830))`

Comment (joaomcteixeira) [reply CITING.md]: Done

**Review by amjjbonvin — APPROVED** [review]

**Review by mgiulini — COMMENTED** [review]
Nice contribution. One thing: when running `tox -e docs` I have several warnings, although the overall command succeeded. These warnings seem somehow related to the naming convention and to the `sphinx.ext.autosectionlabel` parameter. Do you see them? Can they be safely ignored?

An example of warning:
`./haddock3/docs/clients/clicfg.rst:2: WARNING: duplicate label named arguments, other instance in ./haddock3/docs/clients/clibm.rst`

**Comment by joaomcteixeira** [comment]
Yes, I see them. Yes, they can be ignored. Sometimes there are `CRITICALS` for which you have to read them carefully. In the CI however, don't know why yet, lots of errors pop. But I think it's just because things in CI are more verbose. Once we have them in ReadTheDocs, RTD itself has a CI that will link to this repository. So far in all projects I have used this it has been safe.

**Comment by mgiulini** [comment]
OK, perfect. I would maybe create an issue about it, so that we don't forget why these warning exist and that we may find a solution to remove/suppress them. What do you think? 

As for the rest, green light from me, great job!

**Comment by joaomcteixeira** [comment]
Those particular issues you commented about are because of the repeated name arguments in the clients, for example `--logging`. There's no way we are going to solve that. We will just leave it as it is for now.

**Pull request merged**